sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -553,19 +594,29 @@ AllocaCommandBase *Scheduler::GraphBuilder::getOrCreateAllocaForReq(
 
   Requirement FullReq(/*Offset*/ {0, 0, 0}, Req->MMemoryRange,
                       Req->MMemoryRange, access::mode::read_write,
-                      Req->MSYCLMemObj, Req->MDims, Req->MElemSize);
+                      Req->MSYCLMemObj, Req->MDims, Req->MElemSize,
+                      Req->MOffsetInBytes);
 
   Requirement *SearchReq = ForceFullReq ? &FullReq : Req;
 
   AllocaCommandBase *AllocaCmd = findAllocaForReq(Record, SearchReq, Queue);
 
   if (!AllocaCmd) {
     if (!ForceFullReq && IsSuitableSubReq(Req)) {
-      if (Req->MDims != 1)
-        throw runtime_error("OpenCL only supports 1D sub buffers");
-
-      auto *ParentAlloca = getOrCreateAllocaForReq(Record, Req, Queue, true);
+      // Get parent requirement. It's hard to get right parents' range
+      // so full parent requirement has range represented in bytes
+      range<3> ParentRange{Req->MSYCLMemObj->getSize(), 1, 1};
+      Requirement ParentRequirement(/*Offset*/ {0, 0, 0}, ParentRange,
+                                    ParentRange, access::mode::read_write,
+                                    Req->MSYCLMemObj, /*Dims*/ 1,
+                                    /*Working with bytes*/ sizeof(char),
+                                    /*MOffsetInBytes*/ 0);
+
+      auto *ParentAlloca =
+          getOrCreateAllocaForReq(Record, &ParentRequirement, Queue, true);
```

Comment:
Probably it's better to just create alloca command here instead of calling getOrCreateAlloca again.